open_spiel/games/phantom_go_test.cc

```diff
@@ -0,0 +1,231 @@
+    SPIEL_CHECK_FALSE(state.History() == stateClone->History());
+}
+
+void HandicapTest() {
+  std::shared_ptr<const Game> game =
+      LoadGame("phantom_go", {{"board_size", open_spiel::GameParameter(kBoardSize)},
+                      {"komi", open_spiel::GameParameter(kKomi)},
+                      {"handicap", open_spiel::GameParameter(1)}});
+  PhantomGoState state(game, kBoardSize, kKomi, 2);
+  SPIEL_CHECK_EQ(state.CurrentPlayer(), ColorToPlayer(GoColor::kWhite));
+  SPIEL_CHECK_EQ(state.board().PointColor(MakePoint("d4")), GoColor::kBlack);
+  
+}
+
+void IllegalMoveTest()
+{
+    GameParameters params;
+    params["board_size"] = GameParameter(kBoardSize);
+    std::shared_ptr<const Game> game =
+        LoadGame("phantom_go", params);
+    PhantomGoState state(game, kBoardSize, kKomi, 0);
+    SPIEL_CHECK_EQ(state.CurrentPlayer(), ColorToPlayer(GoColor::kBlack));
+    state.ApplyAction(5);
+    SPIEL_CHECK_EQ(state.CurrentPlayer(), ColorToPlayer(GoColor::kWhite));
+    state.ApplyAction(5);
+    SPIEL_CHECK_EQ(state.CurrentPlayer(), ColorToPlayer(GoColor::kWhite));
+}
+
+void StoneCountTest()
+{
+    GameParameters params;
+    params["board_size"] = GameParameter(kBoardSize);
+    std::shared_ptr<const Game> game =
+        LoadGame("phantom_go", params);
+    PhantomGoState state(game, kBoardSize, kKomi, 0);
+    SPIEL_CHECK_EQ(state.board().getStoneCount()[(uint8_t)GoColor::kBlack], 0);
+    SPIEL_CHECK_EQ(state.board().getStoneCount()[(uint8_t)GoColor::kWhite], 0);
+    state.ApplyAction(5);
+    SPIEL_CHECK_EQ(state.board().getStoneCount()[(uint8_t)GoColor::kBlack], 1);
+    SPIEL_CHECK_EQ(state.board().getStoneCount()[(uint8_t)GoColor::kWhite], 0);
+    state.ApplyAction(6);
+    SPIEL_CHECK_EQ(state.board().getStoneCount()[(uint8_t)GoColor::kBlack], 1);
+    SPIEL_CHECK_EQ(state.board().getStoneCount()[(uint8_t)GoColor::kWhite], 1);
+
+}
+
+void ConcreteActionsAreUsedInTheAPI() {
+  std::shared_ptr<const Game> game =
+      LoadGame("phantom_go", {{"board_size", open_spiel::GameParameter(kBoardSize)}});
+  std::unique_ptr<State> state = game->NewInitialState();
+
+  SPIEL_CHECK_EQ(state->NumDistinctActions(), kBoardSize * kBoardSize + 1);
+  SPIEL_CHECK_EQ(state->LegalActions().size(), state->NumDistinctActions());
+  for (Action action : state->LegalActions()) {
+    SPIEL_CHECK_GE(action, 0);
+    SPIEL_CHECK_LE(action, kBoardSize * kBoardSize);
+  }
+}
+
+void ResampleFromInfostateVisualTest()
+{
+    std::cout << "Starting ResampleFromInfostate visual Test\n";
+    GameParameters params;
+    params["board_size"] = GameParameter(kBoardSize);
+    std::shared_ptr<const Game> game =
+        LoadGame("phantom_go", params);
+    PhantomGoState state(game, kBoardSize, kKomi, 0);
+
+    for(int i = 0; i < 150; i++)
+    {
+        std::vector<Action> actions = state.LegalActions();
+        std::shuffle(actions.begin(), actions.end(), std::mt19937(std::random_device()()));
```

Comment:
You can use SampleAction or even better https://www.cplusplus.com/reference/random/uniform_int_distribution/